Minor DRY in Buildkite UI tests

mokagio opened the pull request:
It seems unlikely we'll want to run multiple UI tests jobs from the same build with different schemes and/or iOS versions. So, I centralized the definition of those parameters.

For the record, these are some leftover tweaks I made when experimenting with Test Analytics. I haven't developed a sudden urge to DRY random scripts 😅 (or rather, I do have that urge but I try to keep it in check).

## Testing 

If CI is green, we're good.

## Regression Notes

1. Potential unintended areas of impact – N.A.
2. What I did to test those areas of impact (or what existing automated tests I relied on) – N.A.
3. What automated tests I added (or what prevented me from doing so) – N.A.

---

- [x] I have completed the Regression Notes.
- [x] I have considered adding unit tests for my changes. **N.A.**
- [x] I have considered adding accessibility improvements for my changes. **N.A.**
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary. **N.A.**

mokagio committed:
Remove `try_count` parameter from test commands in CI

That parameter is not part of Fastlane `run_test` (`scan`). It was a
feature provided by a plugin we long since removed, see:

- 75f87334ee967e1ae8c00d8ccfbecef28fef237d
- https://github.com/wordpress-mobile/WordPress-iOS/pull/17456
- https://github.com/wordpress-mobile/WordPress-iOS/pull/17841

The parameter was removed from the CI script at the time, but reappeared
when moving from CircleCI to Buildkite.

mokagio committed:
Stop inject UI tests target name in CI; There's only one to use!

mokagio committed:
Centralize iOS version definition in Buildkite UI tests

mokagio commented on .buildkite/commands/run-ui-tests.sh:
```diff
@@ -36,7 +36,7 @@ echo "--- 🔬 Testing"
 rake mocks &
 set +e
-bundle exec fastlane test_without_building name:"$TEST_NAME" try_count:3 device:"$DEVICE" ios_version:"$IOS_VERSION"
+bundle exec fastlane test_without_building name:WordPressUITests device:"$DEVICE" ios_version:"$IOS_VERSION"
```
Regarding `try_count: 3`: That parameter is not part of Fastlane `run_test` (`scan`). It was a feature provided by a plugin we long since removed, see:

- https://github.com/wordpress-mobile/WordPress-iOS/commit/75f87334ee967e1ae8c00d8ccfbecef28fef237d
- https://github.com/wordpress-mobile/WordPress-iOS/pull/17456
- https://github.com/wordpress-mobile/WordPress-iOS/pull/17841

The parameter was removed from the CI script at the time, but reappeared when moving from CircleCI to Buildkite.

pachlava replied:
It's good to have `try_count` removed, because `WordPressUITests` test plan in Xcode also has `Retry on Failure` set to 3, so now it's potentially less confusing for a person that will try to adjust tests repetitions.

pachlava commented on .buildkite/commands/run-ui-tests.sh:
```diff
@@ -1,10 +1,10 @@
-IOS_VERSION=$3
+DEVICE=$1
+# Default to the latest iOS, with the option of overriding
+IOS_VERSION=${2:-15.0}
```
It's not related to the PR, but it seems like specifying the iOS version in the context of execution on CI VM does not really matter, because the VM comes (from what I have seen) only with a default set of iOSes typical for the used Xcode version. Locally, it's possible to download the Simulators with the other iOS versions, but I don't think this is the realistic case for CI. For the case of Xcode 13.4.1, it comes with iOS 15.5 (this can be seen in the [log here](https://buildkite.com/automattic/wordpress-ios/builds/9686#0182ce8b-69f5-4b39-b608-84916b8a1c9e/783)). 

So, in reality, the tests were executed on 15.5, but it's not very obvious. Maybe it's worth removing the iOS version mention at all, to avoid possible confusion?

mokagio replied:
Good point, thank you for bringing this up.

mokagio replied:
Screenshot showing CI only has access to iOS 15.5 Simulators:

<img width="1126" alt="Untitled2" src="https://user-images.githubusercontent.com/1218433/186592282-ae88a602-c903-4d1d-8e4e-e07e81cf5311.png">


Screenshot showing that, even though `scan` receives the deployment target parameter as 15.0, the `xctestrun` it uses is from a 15.5 build. I wouldn't be surprised if `scan`/`xcodebuild` ignored that 15.0 when using an `xctestrun`.

<img width="1126" alt="Untitled3" src="https://user-images.githubusercontent.com/1218433/186592014-fced9d6b-72e1-4e9b-8aec-03a6805388b3.png">

mokagio replied:
@pachlava, implemented your suggestion in [wordpress-mobile/WordPress-iOS@`1143b31` (#19193)](https://github.com/wordpress-mobile/WordPress-iOS/pull/19193/commits/1143b31c0394b854ec95da87276a5ce266e539d1)

pachlava reviewed (commented):
Thanks, @mokagio 👋 🙂 

The changes look good to me, and now the code looks more tidy. I left only [one note](https://github.com/wordpress-mobile/WordPress-iOS/pull/19193#discussion_r953666294) related to the usage of `IOS_VERSION` parameter - I don't think it's urgent, but maybe it's worth removing it in the long run 🤔 

Because of the above (and because auto-merge is enabled for this PR), I'm not giving an explicit approval, for the case if you think it's appropriate to remove it in this PR.

mokagio committed:
Remove `ios_version` parameter from CI UI tests – It was ignored

See discussion with @pachlava here
https://github.com/wordpress-mobile/WordPress-iOS/pull/19193#discussion_r953666294

I did not remove the option to specify the parameter in Fastlane because
it might still be useful locally.

pachlava reviewed (approved):
Thank you for addressing the iOS version issue, @mokagio 👋  The CI is green, so 🚢
